### DISCUSSION TITLE
Do you have any plans support vue3 suspense？

### DISCUSSION THREAD

**Discussion by jueinin** [DamianOsipiuk/vue-query#12] [Q&A] [answered]
Vue3 is so similar with react。

I don't why they choose vue as project framework. 

we obviously can choose a more mature framework such as react😒😒

**Comment by jueinin** [comment]
Sorry just some complaints😅

**Comment by DamianOsipiuk** [comment, marked as answer]
Hi, I will try to come up with some clean way to support Suspense.
I do not think that the react-query way would work in Vue (although I might be wrong).

I was thinking about something along:
```
async setup() {
  const query = useQuery(...);
  await query.suspense()
}
```
But I need to test that.

**Reply by DamianOsipiuk** [reply]
@jueinin Suspense support added in https://github.com/DamianOsipiuk/vue-query/releases/tag/v1.2.0

**Comment by jueinin** [comment]
any way...  the suspense can not used in Vue2.x. do you have any ideas. i found this [repo](https://shuidi-fed.github.io/vue-async-manager/guide.html#what-does-an-asynchronous-call-mean) but it not ideal to work. 

suspense is a particularly useful feature. i am figure outing to use it in vue2

**Comment by kachidk** [comment]
Sometimes suspense is overkill if you want suspense for a simple API request just make a suspense component and toggle it on or off with state explicitly.

I do that in nextjs since it doesn't support suspense fully yet (experimental)
